src/univariate/discrete/betabinomial.jl

```diff
@@ -0,0 +1,87 @@
+immutable BetaBinomial <: DiscreteUnivariateDistribution
+    n::Int
+    α::Float64
+    β::Float64
+
+    function BetaBinomial(n::Real, α::Real, β::Real)
+        @check_args(BetaBinomial, n >= zero(n) && α >= zero(α) && β >= zero(β))
+        new(n, α, β)
+    end
+end
+
+@distr_support BetaBinomial 0 d.n
+insupport(d::BetaBinomial, x::Real) = 0 <= x <= d.n
+
+#### Parameters
+
+ntrials(d::BetaBinomial) = d.n
+
+params(d::BetaBinomial) = (d.n, d.α, d.β)
+
+#### Properties
+
+mean(d::BetaBinomial) = (d.n * d.α) / (d.α + d.β)
+function var(d::BetaBinomial)
+    n, α, β = d.n, d.α, d.β
+    numerator = n * α * β * (α + β + n)
+    denominator = (α + β)^2.0 * (α + β + 1.0)
+    return numerator / denominator
+end
+
+function skewness(d::BetaBinomial)
+    n, α, β = d.n, d.α, d.β
+    t1 = (α + β + 2 * n) * (β - α) / (α + β + 2)
+    t2 = sqrt((1 + α +β) / (n * α * β * (n + α + β)))
+    return t1 * t2
+end
+
+function kurtosis(d::BetaBinomial)
+  n, α, β = d.n, d.α, d.β
+  alpha_beta_sum = α + β
+  alpha_beta_product = α * β
+  numerator = ((alpha_beta_sum)^2) * (1 + alpha_beta_sum)
+  denominator = (n * alpha_beta_product) * (alpha_beta_sum + 2) * (alpha_beta_sum + 3) * (alpha_beta_sum + n)
+  left = numerator / denominator
+  right = (alpha_beta_sum) * (alpha_beta_sum - 1 + 6*n) + 3 * alpha_beta_product * (n - 2) + 6 * n^2
+  right -= (3 * alpha_beta_product * n * (6 - n)) / alpha_beta_sum
+  right -= (18 * alpha_beta_product * n^2) / (alpha_beta_sum)^2
+  return (left * right) - 3
+end
+
+function pdf(d::BetaBinomial, k::Int)
+    n, α, β = d.n, d.α, d.β
+    choose = binomial(n, k)
+    numerator = beta(k + α, n - k + β)
+    denominator = beta(α, β)
+    return choose * (numerator / denominator)
+end
+
+function pdf(d::BetaBinomial)
+    n, α, β = d.n, d.α, d.β
+    k = 0:n
+    binoms = Float64[binomial(n, i) for i in k]
+    fixed_beta = beta(α, β)
+    return binoms .* beta(k .+ α, n .- k .+ β) ./ fixed_beta
+end
+
+entropy(d::BetaBinomial) = entropy(Categorical(pdf(d)))
+median(d::BetaBinomial) = median(Categorical(pdf(d))) - 1
+mode(d::BetaBinomial) = indmax(pdf(d)) - 1
+modes(d::BetaBinomial) = [x - 1 for x in modes(Categorical(pdf(d)))]
+
+quantile(d::BetaBinomial, p::Float64) = quantile(Categorical(pdf(d)), p) - 1
+
+function cdf(d::BetaBinomial, x::Int)
+  c = 0.0
+  for y = minimum(d):min(floor(Int,x), maximum(d))
+    c += pdf(d, y)
+  end
+
+  # this is needed due to the fact that pdf(d, y) can return irrational values and so due to the impl
+  # above we can end up with cdf values that are basically 1 but are either slightly greater or smaller
+  if abs(c - 1.0) < 1.0e-10
```

Comment:
Digging a little further, the `beta` function in Julia is defined [here] (https://github.com/JuliaLang/julia/blob/86cd5ea3c61426ebddd1d3809ffa869d6726fb30/base/special/gamma.jl#L407-L412) as

```julia
function beta(x::Number, w::Number)
    yx, sx = lgamma_r(x)
    yw, sw = lgamma_r(w)
    yxw, sxw = lgamma_r(x+w)
    return exp(yx + yw - yxw) * (sx*sw*sxw)
end
```

The `lgamma_r` function here is a call to libm (which is usually openlibm from Julia).  If you compare the values returned from this function call with those produced by the log gamma function on Wolfram Alpha, you'll see a small discrepancy in the last few digits (I'd post more details, but I need to head to bed).

Would you be able to open an issue on Julia or openlibm (or both) describing the problem?  That might be the easiest way to get to a better solution here.  (There are other paths that could be taken--e.g., calculating these values using higher precision--but hopefully that would lead to a better solution.)

And that shouldn't hold up this PR.